sys/include/net/gnrc/mac/internal.h

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (C) 2015 Daniel Krebs
+ *               2016 INRIA
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     net_gnrc_mac  A common MAC type for providing key MAC parameters and helper functions
+ * @ingroup     net
+ * @brief       A common MAC type for providing key MAC parameters and helper functions.
+ * @{
+ *
+ * @file
+ * @brief       Interface definition for internal functions of MAC protocol
+ *
+ * @author      Daniel Krebs <github@daniel-krebs.net>
+ * @author      Shuguo Zhuo  <shuguo.zhuo@inria.fr>
+ */
+
+#ifndef GNRC_MAC_INTERNAL_H_
+#define GNRC_MAC_INTERNAL_H_
+
+#include <stdint.h>
+#include <net/gnrc/mac/types.h>
+#include <net/ieee802154.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/* @brief Extract the destination address out of an GNRC_NETTYPE_NETIF pktsnip
+ *
+ * @param[in]   pkt                 pktsnip from whom to extract
+ * @param[out]  pointer_to_addr     pointer to address will be stored here
+ *
+ * @return                          length of destination address
+ */
+int gnrc_mac_get_dstaddr(gnrc_pktsnip_t* pkt, uint8_t* pointer_to_addr[]);
```

Comment:
I don't know exactly what it was, but there was a problem with array parameters. Please use `uint8_t **` pointer instead